elyra/pipeline/kfp/kfp_authentication.py

```diff
@@ -0,0 +1,739 @@
+                                      provider=self._type)
+
+        with requests.Session() as s:
+
+            request_history = []
+
+            ################
+            # Determine if Endpoint is Secured
+            ################
+            resp = s.get(kf_endpoint, allow_redirects=True)
+            request_history.append((kf_endpoint, resp))
+            if resp.status_code != HTTPStatus.OK:
+                raise AuthenticationError(f'Error detecting whether Kubeflow server at {kf_endpoint} is secured: '
+                                          f'HTTP status code {resp.status_code}'
+                                          f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            if len(resp.history) == 0:
+                # if we were NOT redirected, then the endpoint is UNSECURED
+                # treat this as an error.
+                raise AuthenticationError(f'The Kubeflow server at {kf_endpoint} is not secured using DEX with LDAP. '
+                                          f'Update the authentication type in runtime configuration '
+                                          f'\'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            ################
+            # Get Dex Login URL
+            ################
+            redirect_url_obj = urlsplit(resp.url)
+
+            # if we are at `/auth?=xxxx` path, we need to select
+            # the LDAP auth type
+            if re.search(r"/auth$", redirect_url_obj.path):
+                redirect_url_obj = redirect_url_obj._replace(
+                    path=re.sub(r"/auth$", "/auth/ldap", redirect_url_obj.path)
+                )
+            else:
+                # verify that KF is secured by LDAP
+                m = re.search(r"/auth/([^/]*)/?", redirect_url_obj.path)
+                if m and m.group(1) != 'ldap':
+                    raise AuthenticationError(
+                        f'The Kubeflow server at {kf_endpoint} redirected to an unexpected HTTP path '
+                        f'(\'{redirect_url_obj.path}\'). Verify that Kubeflow is configured for \'{self._type.name}\''
+                        f' and, if necessary, update the authentication type in runtime configuration '
+                        f'\'{runtime_config_name}\'.',
+                        provider=self._type,
+                        request_history=request_history)
+
+            # if we are at `/auth/xxxx/login` path, then no further action is needed
+            # (we can use it for login POST)
+            if re.search(r"/auth/.*/login$", redirect_url_obj.path):
```

Comment:
This check should not be for arbitrary `/auth/xxxx/login` but for `/auth/ldap/login`